smart-contracts/contracts/mixins/MixinLockMetadata.sol

```diff
@@ -31,7 +31,7 @@ contract MixinLockMetadata is
   string private lockSymbol;
 
   // the base Token URI for this Lock. If not set by lock owner, the global URI stored in Unlock is used.
-  string private baseTokenURI;
+  string public baseTokenURI;
```

Comment:
@NickCuso Although this is the simplest way to expose this data publicly, Julien would like the function to fall back to getting the `globalBaseTokenURI` from Unlock if this is not set. I'll make the changes to implement that for v8, unless you have an idea of how we can do this with extensions?

Reply:
Hmm we could create an extension that reads the tokenUri and removes the address and token to extract the base uri. But it's probably better to build this for v8 as that's a little hacky.

On a related note, it seems that if the lock has a base uri the token uri shouldn't also add the lock's address. That may be redundant for some use cases and if it is needed, the lock's address could be included when setting the uri

Reply:
Hmm. maybe so. But in order to guarantee that each key has a globally unique tokenURI (which I think we should ensure), the lock addresses seems to be the simplest way to do this. 
I guess it would be fine to omit the lock's address when a custom URI has been set, as long as each lock has a unique custom URI. Most of the time this would probably be the case.